### DISCUSSION TITLE
请教下速度慢如何排查（ROS 防火墙设置问题）

### DISCUSSION THREAD

**Discussion by zreo0** [kkkgo/PaoPaoGateWay#48] [问和答] [answered]
感谢大佬提供的很方便的 DNS 和 网关，尝试部署了一下，比之前自己折腾方便多了！

求教一个问题，在部署完之后，我发现冲浪的速度有所下降，同样的节点，直接使用 surge 的时候基本页面都是秒开的，使用 dns + gateway 的配置后，通是通的，但是明显感觉到速度下降了很多，好几秒才能打开同样的页面（比如 Google 和 YouTube）。
我的配置方式如下：
一台J4125上使用 PVE 部署了 PaoPaoDNS（在一个 Linux 虚拟机中使用 Docker 部署），然后 PVE 直接一个虚拟机启动的 PaoPaoGateway，配置文件通过内网到 PaoPaoDNS 获取。
网络主路由是一台 MikroTik，把 DNS 配置成 PaoPaoDNS 的 ip，DHCP 中也将下发的 DNS 改为 PaoPaoDNS。同时 ROS 中新增了一条静态路由将 7.0.0.0/8 转发到 PaoPaoGateway。

在测试过程中梯子的多个节点都试过，也不像是不稳定的问题。是部署方式的问题吗？有啥办法能进行排查下？感谢！

**Comment by kkkgo** [comment]
有可能是你节点UDP支持的问题或者浏览器DNS缓存更新不及时的问题。
可以尝试定制SNIFF=yes的镜像。

**Comment by kkkgo** [comment]
另外ROS可以参考这位 https://github.com/kkkgo/PaoPaoGateWay/discussions/45

**Reply by kkkgo** [reply]
里面提到添加Filter Rules，可以参考试试。

**Reply by zreo0** [reply]
好的，我尝试一下，感谢。

**Reply by zreo0** [reply, marked as answer]
确实是 ROS 默认防火墙的问题，参考[这篇文章](https://zhelin.me/post/e4aba903f3826e7b/)其中的部分得到解决，记录下解决方案：
> 打开 IP > Firewall > Filter Rules
找到 action=drop chain=forward comment=“drop invalid” connection-state=invalid 这一条，将 General 下的 In. Interface List 改为 !LAN 防止防火墙将加密过后的流量标记为 invalid 而造成 TCP 流量握手缓慢
